### ISSUE TITLE
Add name_regex argument for data source aws_iam_user (as in data_source aws_iam_roles)

### ISSUE THREAD

**Issue by evairmarinho** [hashicorp/terraform-provider-aws#20717]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
To find users on AWS, it would be interesting to search through regex. Example: "my-company-*" .

### New or Affected Resource(s)

Data Source
* aws_iam_user

### Potential Terraform Configuration

```hcl
data "aws_iam_user" "my_company_users"{
name_regex = "my-company-*"
}

```

### References

Name regex

* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_roles#roles-filtered-by-name-regex

**Comment by kamilturek** [comment]
Data source `aws_iam_users` with `name_regex` attribute is available since [3.60](https://github.com/hashicorp/terraform-provider-aws/pull/20877https://github.com/hashicorp/terraform-provider-aws/pull/20877).

**Comment by evairmarinho** [comment]
Thanks @kamilturek ! I will close this issue.
